File: summer/WebContent/js/system/windows/controls/primitives/GeneratorStatus.js
/**
 * from IItemContainerGenerator
 */ 

/**
 * GeneratorStatus
 */
define(["dojo/_base/declare"], function(declare){

    var GeneratorStatus=declare("GeneratorStatus", null,{ 
    });	
    
    GeneratorStatus.NotStarted = 0;
    GeneratorStatus.GeneratingContainers = 1;
    GeneratorStatus.ContainersGenerated = 2;
    GeneratorStatus.Error = 3;

    return GeneratorStatus;
    
});

File: org.summer.view.widget/src/org/summer/view/widget/controls/primitives/GeneratorStatus.java
package org.summer.view.widget.controls.primitives;

/// <summary> 
/// This enum is used by the ItemContainerGenerator to indicate its status.
/// </summary> 
public enum GeneratorStatus {
	// /<summary>The generator has not tried to generate content</summary>
	NotStarted,
	// /<summary>The generator is generating containers</summary>
	GeneratingContainers,
	// /<summary>The generator has finished generating containers</summary>
	ContainersGenerated,
	// /<summary>The generator has finished generating containers, but
	// encountered one or more errors</summary>
	Error
}